tools/c7n_azure/c7n_azure/resources/storage.py

```diff
@@ -64,156 +65,120 @@ class resource_type(ArmResourceManager.resource_type):
+     - Azure Logging and Metrics services
+     - Two specific IPs
+     - Two subnets
 
-            filters:
-                - type: value
-                  key: properties.networkAcls.ipRules
-                  value_type: size
-                  op: eq
-                  value: 0
-
-            actions:
-                - type: set-firewall-rules
-                  bypass-rules:
-                      - Logging
-                      - Metrics
-                  ip-rules:
-                      - 11.12.13.0/16
-                      - 21.22.23.24
-                  virtual-network-rules:
-                      - <subnet_resource_id>
-                      - <subnet_resource_id>
+     .. code-block:: yaml
 
-    """
+         policies:
+             - name: add-storage-firewall
+               resource: azure.storage
+
+             filters:
+                 - type: value
+                   key: properties.networkAcls.ipRules
+                   value_type: size
+                   op: eq
+                   value: 0
+
+             actions:
+                 - type: set-firewall-rules
+                   bypass-rules:
+                       - Logging
+                       - Metrics
+                   ip-rules:
+                       - 11.12.13.0/16
+                       - 21.22.23.24
+                   virtual-network-rules:
+                       - <subnet_resource_id>
+                       - <subnet_resource_id>
+
+     """
 
     schema = type_schema(
         'set-firewall-rules',
-        required=[],
+        rinherit=SetFirewallAction.schema,
         **{
             'default-action': {'enum': ['Allow', 'Deny'], "default": 'Deny'},
-            'append': {'type': 'boolean', "default": False},
             'bypass-rules': {'type': 'array', 'items': {
                 'enum': ['AzureServices', 'Logging', 'Metrics']}},
-            'ip-rules': {'type': 'array', 'items': {'type': 'string'}},
-            'virtual-network-rules': {'type': 'array', 'items': {'type': 'string'}}
         }
     )
 
     def __init__(self, data, manager=None):
-        super(StorageSetNetworkRulesAction, self).__init__(data, manager)
+        super(StorageSetFirewallAction, self).__init__(data, manager)
         self._log = logging.getLogger('custodian.azure.storage')
         self.rule_limit = 200
 
-    def _prepare_processing(self):
-        self.client = self.manager.get_client()
-        self.append = self.data.get('append', False)
-
     def _process_resource(self, resource):
-        rules = self._build_ip_rules(resource, self.data.get('ip-rules', []))
-
         # Build out the ruleset model to update the resource
         rule_set = NetworkRuleSet(default_action=self.data.get('default-action', 'Deny'))
 
+        # Add IP rules
+        existing_ip = resource['properties']['networkAcls'].get('ipRules', [])
+        rules = self._build_ip_rules(existing_ip, self.data.get('ip-rules', []))
```

Comment:
nit: ip_rules. There are so many rules and it's difficult to keep track of them without good names.

Reply:
There were so many rules it took me a while to realize what you meant, but yes I agree :)  I am adding SQL now and will update them all.